algorithms/src/main/java/com/aparamonov/algostructure/search/BFS.java

```diff
@@ -14,8 +14,8 @@
 
     public List<Node<T>> search(Node<T> node) {
 
-        List<Node<T>> visited = new LinkedList<>();
-        List<Node<T>> unvisited = new LinkedList<>();
+        List<Node<T>> visited = new ArrayList<>();
```

Comment:
i'm not really sure this change is necessary. you're working with *graph* structures, not arrays. please check the performance for linked and array-lists